In [1]:
!ls dataset

ClaimFake_combined.csv		       NewsFake_combined.csv
ClaimFake_tweets_combined.csv	       NewsFake_tweets_combined.csv
ClaimFake_tweets_replies_combined.csv  NewsFake_tweets_replies_combined.csv
ClaimReal_combined.csv		       NewsReal_combined.csv
ClaimReal_tweets_combined.csv	       NewsReal_tweets_combined.csv
ClaimReal_tweets_replies_combined.csv  NewsReal_tweets_replies_combined.csv


In [2]:
DATAPATH = 'dataset/'

In [3]:
import pandas as pd

In [4]:
C_Fake = pd.read_csv(DATAPATH + 'ClaimFake_combined.csv')
C_Fake_Edge = pd.read_csv(DATAPATH + 'ClaimFake_tweets_replies_combined.csv')

In [5]:
display(C_Fake.head(5))
display(C_Fake_Edge.head(5))

Unnamed: 0        fact_check_url  \
0      100000  medicalnewstoday.com   
1      100001  medicalnewstoday.com   
2      100002  medicalnewstoday.com   
3      100003  medicalnewstoday.com   
4      100004  medicalnewstoday.com   

                                            news_url  \
0  https://www.medicalnewstoday.com/articles/coro...   
1  https://www.medicalnewstoday.com/articles/coro...   
2  https://www.medicalnewstoday.com/articles/coro...   
3  https://www.medicalnewstoday.com/articles/coro...   
4  https://www.medicalnewstoday.com/articles/coro...   

                                               title  
0  "Spraying chlorine or alcohol on the skin kill...  
1   "Only older adults and young people are at risk"  
2                     "Children cannot get COVID-19"  
3                    "COVID-19 is just like the flu"  
4                      "Everyone with COVID-19 dies"

news_id             tweet_id             reply_id
0   100001  1242474839966765056  1242474835512446976
1   100001  1242474839966765056  1242474843137703938
2   100001  1242400142335983619  1242388627943878657
3   100001  1242400142335983619  1242400453700096001
4   100003  1248606541793488896  1248609022841729028

In [6]:
N_Fake = pd.read_csv(DATAPATH + 'NewsFake_combined.csv')
N_Fake_Edge = pd.read_csv(DATAPATH + 'NewsFake_tweets_replies_combined.csv')

In [7]:
display(N_Fake.head(5))
display(N_Fake_Edge.head(5))

Unnamed: 0     type                                     fact_check_url  \
0           0     post  https://factcheck.afp.com/false-advice-refusin...   
1           1  article  https://www.politifact.com/factchecks/2020/apr...   
2           2     post  https://factcheck.afp.com/posts-claim-children...   
3           3     post  https://checkyourfact.com/2020/04/20/fact-chec...   
4           5     post  https://checkyourfact.com/2020/04/20/fact-chec...   

                      archive  \
0  https://perma.cc/J4N6-39D5   
1                         NaN   
2  https://perma.cc/V4HX-M2XJ   
3                         NaN   
4                         NaN   

                                            news_url news_url2 news_url3  \
0  https://www.facebook.com/photo.php?fbid=551960...       NaN       NaN   
1  http://legis.wisconsin.gov/assembly/republican...       NaN       NaN   
2  https://www.facebook.com/iAmJessenia/photos/a....       NaN       NaN   
3  https://www.facebook.com/kokernagnews/photos/a...       NaN       NaN   
4  https://www.facebook.com/photo.php?fbid=102193...       NaN       NaN   

  news_url4 news_url5                                              title  \
0       NaN       NaN  Facebook posts shared in at least three countr...   
1       NaN       NaN  Wisconsin is Òclearly seeing a decline in COVI...   
2       NaN       NaN  Facebook posts claim a child who is infected w...   
3       NaN       NaN  IndiaÕs Ministry of Home Affairs banning citiz...   
4       NaN       NaN  42 Democratic senators, plus two Independents,...   

                                           newstitle  \
0                                                NaN   
1  "Wisconsin Legislature Takes Gov. Evers to Court"   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                             content abstract publish_date  \
0                                                NaN      NaN          NaN   
1  speaker robin vos r rochester and senate major...      NaN          NaN   
2                                                NaN      NaN          NaN   
3                                                NaN      NaN          NaN   
4                                                NaN      NaN          NaN   

  meta_keywords  
0           NaN  
1            ""  
2           NaN  
3           NaN  
4           NaN

news_id             tweet_id             reply_id
0       17  1255484783884369921  1254913226325872642
1       17  1255484783884369921  1254985211265159169
2       17  1255484783884369921  1255484999203258370
3       26  1249534887691984897  1249535327829712898
4       26  1249534887691984897  1249536037489184769

In [8]:
!ls dataset

ClaimFake_combined.csv		       NewsFake_combined.csv
ClaimFake_tweets_combined.csv	       NewsFake_tweets_combined.csv
ClaimFake_tweets_replies_combined.csv  NewsFake_tweets_replies_combined.csv
ClaimReal_combined.csv		       NewsReal_combined.csv
ClaimReal_tweets_combined.csv	       NewsReal_tweets_combined.csv
ClaimReal_tweets_replies_combined.csv  NewsReal_tweets_replies_combined.csv


In [9]:
C_Real = pd.read_csv(DATAPATH + 'ClaimReal_combined.csv')
C_Real_Edge = pd.read_csv(DATAPATH + 'ClaimReal_tweets_replies_combined.csv')

In [10]:
display(N_Fake.head(5))
display(N_Fake_Edge.head(5))

Unnamed: 0     type                                     fact_check_url  \
0           0     post  https://factcheck.afp.com/false-advice-refusin...   
1           1  article  https://www.politifact.com/factchecks/2020/apr...   
2           2     post  https://factcheck.afp.com/posts-claim-children...   
3           3     post  https://checkyourfact.com/2020/04/20/fact-chec...   
4           5     post  https://checkyourfact.com/2020/04/20/fact-chec...   

                      archive  \
0  https://perma.cc/J4N6-39D5   
1                         NaN   
2  https://perma.cc/V4HX-M2XJ   
3                         NaN   
4                         NaN   

                                            news_url news_url2 news_url3  \
0  https://www.facebook.com/photo.php?fbid=551960...       NaN       NaN   
1  http://legis.wisconsin.gov/assembly/republican...       NaN       NaN   
2  https://www.facebook.com/iAmJessenia/photos/a....       NaN       NaN   
3  https://www.facebook.com/kokernagnews/photos/a...       NaN       NaN   
4  https://www.facebook.com/photo.php?fbid=102193...       NaN       NaN   

  news_url4 news_url5                                              title  \
0       NaN       NaN  Facebook posts shared in at least three countr...   
1       NaN       NaN  Wisconsin is Òclearly seeing a decline in COVI...   
2       NaN       NaN  Facebook posts claim a child who is infected w...   
3       NaN       NaN  IndiaÕs Ministry of Home Affairs banning citiz...   
4       NaN       NaN  42 Democratic senators, plus two Independents,...   

                                           newstitle  \
0                                                NaN   
1  "Wisconsin Legislature Takes Gov. Evers to Court"   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                             content abstract publish_date  \
0                                                NaN      NaN          NaN   
1  speaker robin vos r rochester and senate major...      NaN          NaN   
2                                                NaN      NaN          NaN   
3                                                NaN      NaN          NaN   
4                                                NaN      NaN          NaN   

  meta_keywords  
0           NaN  
1            ""  
2           NaN  
3           NaN  
4           NaN

news_id             tweet_id             reply_id
0       17  1255484783884369921  1254913226325872642
1       17  1255484783884369921  1254985211265159169
2       17  1255484783884369921  1255484999203258370
3       26  1249534887691984897  1249535327829712898
4       26  1249534887691984897  1249536037489184769

In [11]:
N_Real = pd.read_csv(DATAPATH + 'NewsReal_combined.csv')
N_Real_Edge = pd.read_csv(DATAPATH + 'NewsReal_tweets_replies_combined.csv')

In [12]:
display(N_Real.head(5))
display(N_Real_Edge.head(5))

Unnamed: 0     type  fact_check_url  \
0           0  article  healthline.com   
1           1  article  healthline.com   
2           2  article  healthline.com   
3           3  article  healthline.com   
4           4  article  healthline.com   

                                            news_url  \
0  https://www.healthline.com/health-news/coronav...   
1  https://www.healthline.com/health-news/screen-...   
2  https://www.healthline.com/health-news/heres-e...   
3  https://www.healthline.com/health-news/heres-w...   
4  https://www.healthline.com/health-news/why-you...   

                                               title  \
0                 Coronavirus Outbreak: Live Updates   
1  Screen Time Doesn’t Hurt Kids’ Social Skills, ...   
2  Here’s Exactly Where We Are with Vaccines and ...   
3  Here’s What Happens to the Body After Contract...   
4  Why You Should Be Skeptical of Early Results o...   

                                           newstitle  \
0              "Coronavirus Outbreak: Daily Updates"   
1  "Screen Time Doesn't Hurt Kids' Social Skills,...   
2  "Where We're at with Vaccines and Treatments f...   
3  "Here's What Happens to the Body After Contrac...   
4  "Why You Should Be Skeptical of Leaked Results...   

                                             content  \
0  experts warn most states that are reopening st...   
1  a new study finds an increase in screen time d...   
2  scientists around the world are working on a n...   
3  coronaviruses cause respiratory illnesses so t...   
4  medical experts are studying if remdesivir can...   

                                            abstract publish_date  \
0  the new coronavirus disease outbreak first ide...          NaN   
1  new research found that school age children in...          NaN   
2  scientists around the world are working on a n...          NaN   
3  heres what we know so far about how covid-19 a...          NaN   
4  with no approved treatment specific for covid-...          NaN   

                                       meta_keywords  
0  "coronavirus outbreak, viral outbreak, viral i...  
1  "screen time, child development, smartphones, ...  
2  "covid-19 coronavirus outbreak research progre...  
3  "coronavirus, viral infection, viral outbreak,...  
4  "covid-19, covornavirus, novel coronavirus, vi...

news_id             tweet_id             reply_id
0        0  1253800268300455936  1253801366327300098
1        0  1253800268300455936  1253801454126665729
2        0  1253800268300455936  1253801598985396224
3        0  1253800268300455936  1253801654882893824
4        0  1253800268300455936  1253802430967341057

C_Fake
C_Fake_Edge

C_Real
C_Real_Edge

N_Fake
N_Fake_Edge

N_Real
N_Real_Edge

In [14]:
C_Real.rename(columns={'Unnamed: 0':'news_id'},inplace=True)
N_Real.rename(columns={'Unnamed: 0':'news_id'},inplace=True)

C_Fake.rename(columns={'Unnamed: 0':'news_id'},inplace=True)
N_Fake.rename(columns={'Unnamed: 0':'news_id'},inplace=True)

## Fake

In [17]:
C_Fake['news_id']

0     100000
1     100001
2     100002
3     100003
4     100004
5     100005
6     100006
7     100007
8     100008
9     100009
10    100010
11    100011
12    100012
13    100013
14    100014
15    100015
16    100016
17    100017
18    100018
19    100019
20    100020
21    100021
22    100022
23    100023
24    100024
25    100025
26    100026
27    100027
Name: news_id, dtype: int64

In [16]:
C_Fake_Edge['news_id']

0      100001
1      100001
2      100001
3      100001
4      100003
        ...  
621    100022
622    100022
623    100027
624    100027
625    100027
Name: news_id, Length: 626, dtype: int64

In [18]:
N_Fake['news_id']

0        0
1        1
2        2
3        3
4        5
      ... 
920    971
921    972
922    973
923    974
924    975
Name: news_id, Length: 925, dtype: int64

In [19]:
N_Fake_Edge['news_id']

0        17
1        17
2        17
3        26
4        26
       ... 
7472    968
7473    968
7474    968
7475    968
7476    968
Name: news_id, Length: 7477, dtype: int64

## Real

In [21]:
C_Real['news_id']

0      100000
1      100001
2      100002
3      100003
4      100004
        ...  
485    100369
486    100370
487    100371
488    100372
489    100373
Name: news_id, Length: 490, dtype: int64

In [22]:
C_Real_Edge['news_id']

0        100000
1        100000
2        100001
3        100001
4        100001
          ...  
12653    100367
12654    100367
12655    100367
12656    100367
12657    100338
Name: news_id, Length: 12658, dtype: int64

In [23]:
N_Real['news_id']

0           0
1           1
2           2
3           3
4           4
        ...  
4527    10078
4528    10079
4529    10080
4530    10081
4531    10082
Name: news_id, Length: 4532, dtype: int64

In [24]:
N_Real_Edge['news_id']

0            0
1            0
2            0
3            0
4            0
          ... 
115111    4311
115112    4311
115113    4311
115114    4311
115115    4311
Name: news_id, Length: 115116, dtype: int64

## Data description


#Real 뉴스 ; 490 + 4,532 -> 5,022

C_real ; 490
N_real ; 4,532

#Fake 뉴스 ; 28 + 925 -> 953

C_fake ; 28
N_fake ; 925



#Real 뉴스 전파 ; 12658 + 115116 -> 127,774

C_real_edge ; 12,658
N_real_edge ; 115,116

#Fake 뉴스 전파 626 + 7477 -> 8,103

C_fake_edge ; 626
N_fake_edge ; 7,477

## Combined

label 1 ; Fake , label 0 ; Real

In [27]:
Real = pd.concat([N_Real, C_Real],axis=0)
Fake = pd.concat([N_Fake, C_Fake],axis=0)

In [28]:
Real['label'] = 0
Fake['label'] = 1

In [133]:
regex = re.compile(r'http(s)?:\/\/.*?\/')

### method 1

In [392]:
len(N_Fake['fact_check_url'])

925

In [ ]:
site = []
for i in range(1,953):
    site.append(re.search('https?://(?P<site_name>.*?)/', test_3[i])[1])

In [429]:
url_2 = N_Fake['fact_check_url']
site2 = []
for i in range(len(N_Fake['fact_check_url'])):
    site2.append(re.search('https?://(?P<site_name>.*?)/', url_3[i]))    

In [437]:
site2[1][1]

'www.politifact.com'

In [446]:
url_filtered = []
for i in range(len(site2)):
    url_filtered.append(site2[i][1])

In [448]:
N_Fake['filtered'] = url_filtered

In [447]:
url_filtered

['factcheck.afp.com',
 'www.politifact.com',
 'factcheck.afp.com',
 'checkyourfact.com',
 'checkyourfact.com',
 'www.facebook.com',
 'www.factcheck.org',
 'www.politifact.com',
 'www.politifact.com',
 'www.politifact.com',
 'www.politifact.com',
 'www.washingtonpost.com',
 'checkyourfact.com',
 'checkyourfact.com',
 'www.politifact.com',
 'healthfeedback.org',
 'www.washingtonpost.com',
 'twitter.com',
 'leadstories.com',
 'checkyourfact.com',
 'factcheck.afp.com',
 'leadstories.com',
 'www.politifact.com',
 'leadstories.com',
 'healthfeedback.org',
 'www.politifact.com',
 'www.politifact.com',
 'checkyourfact.com',
 'www.factcheck.org',
 'leadstories.com',
 'www.politifact.com',
 'www.washingtonpost.com',
 'checkyourfact.com',
 'www.factcheck.org',
 'www.politifact.com',
 'www.politifact.com',
 'www.politifact.com',
 'www.politifact.com',
 'leadstories.com',
 'www.politifact.com',
 'www.politifact.com',
 'www.politifact.com',
 'leadstories.com',
 'leadstories.com',
 'factcheck.afp.com

In [439]:
len(site2)

925

In [450]:
set(N_Fake['filtered'])

{'africacheck.org',
 'checkyourfact.com',
 'factcheck.afp.com',
 'healthfeedback.org',
 'hoax-alert.leadstories.com',
 'leadstories.com',
 'pesacheck.org',
 'twitter.com',
 'www.facebook.com',
 'www.factcheck.org',
 'www.instagram.com',
 'www.mythdetector.ge',
 'www.politifact.com',
 'www.thejournal.ie',
 'www.vishvasnews.com',
 'www.washingtonpost.com',
 'www.youtube.com',
 'youtu.be'}

In [460]:
N_Fake['filtered']

0       factcheck.afp.com
1      www.politifact.com
2       factcheck.afp.com
3       checkyourfact.com
4       checkyourfact.com
              ...        
920    healthfeedback.org
921    healthfeedback.org
922     www.factcheck.org
923    healthfeedback.org
924    healthfeedback.org
Name: filtered, Length: 925, dtype: object

In [ ]:
N_Fake['filtered'].replace('youtu.be', 'www.youtube.com')

In [482]:
N_Fake['filtered'].unique()

array(['factcheck.afp.com', 'www.politifact.com', 'checkyourfact.com',
       'www.facebook.com', 'www.factcheck.org', 'www.washingtonpost.com',
       'healthfeedback.org', 'twitter.com', 'leadstories.com',
       'pesacheck.org', 'www.youtube.com', 'www.instagram.com',
       'hoax-alert.leadstories.com', 'africacheck.org',
       'www.vishvasnews.com', 'www.thejournal.ie', 'www.mythdetector.ge'],
      dtype=object)

In [485]:
N_Fake.isnull().sum()

news_id             0
type               40
fact_check_url      0
archive           548
news_url           40
news_url2         832
news_url3         865
news_url4         891
news_url5         910
title               0
newstitle         458
content           509
abstract          656
publish_date      780
meta_keywords     458
filtered            0
dtype: int64

In [486]:
C_Fake.isnull().sum()

news_id           0
fact_check_url    0
news_url          0
title             0
dtype: int64

In [488]:
NewsFake = N_Fake[['news_id','filtered','title']]

In [499]:
ClaimFake.rename(columns={'fact_check_url':'filtered'},inplace=True)

/mnt/a/anaconda3/envs/graph/lib/python3.8/site-packages/pandas/core/frame.py:4441: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().rename(


In [491]:
ClaimFake = C_Fake[['news_id', 'fact_check_url','title']]

In [501]:
Fake = pd.concat([NewsFake, ClaimFake], axis=0)

### Real

In [507]:
C_Real['fact_check_url'].unique()

array(['who.int', 'twitter.com'], dtype=object)

In [566]:
N_Real['fact_check_url']

0                                          healthline.com
1                                          healthline.com
2                                          healthline.com
3                                          healthline.com
4                                          healthline.com
                              ...                        
4527    https://www.politifact.com/factchecks/2020/oct...
4528    https://www.politifact.com/factchecks/2020/oct...
4529    https://www.politifact.com/factchecks/2020/oct...
4530    https://www.politifact.com/factchecks/2020/sep...
4531    https://www.politifact.com/factchecks/2020/sep...
Name: fact_check_url, Length: 4532, dtype: object

In [572]:
site3 = []
for i, checking in enumerate(N_Real['fact_check_url']):
    site3.append(re.search('https?://(?P<site_name>.*?)/', N_Real['fact_check_url'][i]))
    if site3[i] is not None:
        N_Real['fact_check_url'][i] = 'politifact.com'

<ipython-input-572-4205f9b381fa>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  N_Real['fact_check_url'][i] = 'politifact.com'


In [574]:
N_Real['fact_check_url'].unique()

array(['healthline.com', 'sciencedaily.com', 'nih.gov',
       'medicalnewstoday.com', 'mayoclinic.org', 'clevelandclinic.org',
       'webmd.com', 'cdc.gov', 'who.int', 'politifact.com'], dtype=object)

In [583]:
ClaimReal = C_Real[['news_id', 'fact_check_url','title']]
NewsReal = N_Real[['news_id','fact_check_url','title']]

In [584]:
Real = pd.concat([NewsReal, ClaimReal], axis=0)

In [593]:
Fake.rename(columns={'filtered':'fact_check_url'},inplace=True)

In [595]:
Fake['label'] = 0

In [596]:
Real['label'] = 1

### result

In [599]:
Fake.to_csv('Fake.csv')

In [597]:
Fake

news_id        fact_check_url  \
0         0     factcheck.afp.com   
1         1    www.politifact.com   
2         2     factcheck.afp.com   
3         3     checkyourfact.com   
4         5     checkyourfact.com   
..      ...                   ...   
23   100023  medicalnewstoday.com   
24   100024  medicalnewstoday.com   
25   100025  medicalnewstoday.com   
26   100026  medicalnewstoday.com   
27   100027  medicalnewstoday.com   

                                                title  label  
0   Facebook posts shared in at least three countr...      0  
1   Wisconsin is Òclearly seeing a decline in COVI...      0  
2   Facebook posts claim a child who is infected w...      0  
3   IndiaÕs Ministry of Home Affairs banning citiz...      0  
4   42 Democratic senators, plus two Independents,...      0  
..                                                ...    ...  
23   "The outbreak began because people ate bat soup"      0  
24                       "5G helps SARS-CoV-2 spread"      0  
25   "Drinking alcohol reduces the risk of infection"      0  
26  "Injecting or consuming bleach or disinfectant...      0  
27        You can catch coronavirus in swimming pools      0  

[953 rows x 4 columns]

In [603]:
Fake['fact_check_url'].isnull().sum()

0

In [607]:
display(Fake['fact_check_url'].nunique())
display(Fake['fact_check_url'].unique())

18

array(['factcheck.afp.com', 'www.politifact.com', 'checkyourfact.com',
       'www.facebook.com', 'www.factcheck.org', 'www.washingtonpost.com',
       'healthfeedback.org', 'twitter.com', 'leadstories.com',
       'pesacheck.org', 'www.youtube.com', 'www.instagram.com',
       'hoax-alert.leadstories.com', 'africacheck.org',
       'www.vishvasnews.com', 'www.thejournal.ie', 'www.mythdetector.ge',
       'medicalnewstoday.com'], dtype=object)

In [600]:
Real.to_csv('Real.csv')

In [602]:
Real['fact_check_url'].isnull().sum()

0

In [606]:
display(Real['fact_check_url'].nunique())
display(Real['fact_check_url'].unique())

11

array(['healthline.com', 'sciencedaily.com', 'nih.gov',
       'medicalnewstoday.com', 'mayoclinic.org', 'clevelandclinic.org',
       'webmd.com', 'cdc.gov', 'who.int', 'politifact.com', 'twitter.com'],
      dtype=object)

### Real

In [612]:
Real

news_id  fact_check_url  \
0          0  healthline.com   
1          1  healthline.com   
2          2  healthline.com   
3          3  healthline.com   
4          4  healthline.com   
..       ...             ...   
485   100369         who.int   
486   100370         who.int   
487   100371         who.int   
488   100372         who.int   
489   100373         who.int   

                                                 title  label  
0                   Coronavirus Outbreak: Live Updates      1  
1    Screen Time Doesn’t Hurt Kids’ Social Skills, ...      1  
2    Here’s Exactly Where We Are with Vaccines and ...      1  
3    Here’s What Happens to the Body After Contract...      1  
4    Why You Should Be Skeptical of Early Results o...      1  
..                                                 ...    ...  
485  How likely is it that safe, effective vaccines...      1  
486  How quickly could COVID-19 vaccines stop the p...      1  
487  What types of COVID-19 vaccines are being deve...      1  
488  Will other vaccines help protect me from COVID...      1  
489  Will COVID-19 vaccines provide long-term prote...      1  

[5022 rows x 4 columns]

In [617]:
display(N_Real_Edge.head(5))
display(C_Real_Edge.head(5))

news_id             tweet_id             reply_id
0        0  1253800268300455936  1253801366327300098
1        0  1253800268300455936  1253801454126665729
2        0  1253800268300455936  1253801598985396224
3        0  1253800268300455936  1253801654882893824
4        0  1253800268300455936  1253802430967341057

news_id             tweet_id             reply_id
0   100000  1253283636843089920  1253282699688771584
1   100000  1253283636843089920  1253285820485767169
2   100001  1253959004436467713  1253791593179611141
3   100001  1253959004436467713  1253802862372704261
4   100001  1253959004436467713  1253811649498230784

In [622]:
print(len(N_Real_Edge))
print(len(C_Real_Edge))

115116
12658


In [630]:
Real

news_id  fact_check_url  \
0          0  healthline.com   
1          1  healthline.com   
2          2  healthline.com   
3          3  healthline.com   
4          4  healthline.com   
..       ...             ...   
485   100369         who.int   
486   100370         who.int   
487   100371         who.int   
488   100372         who.int   
489   100373         who.int   

                                                 title  label  
0                   Coronavirus Outbreak: Live Updates      1  
1    Screen Time Doesn’t Hurt Kids’ Social Skills, ...      1  
2    Here’s Exactly Where We Are with Vaccines and ...      1  
3    Here’s What Happens to the Body After Contract...      1  
4    Why You Should Be Skeptical of Early Results o...      1  
..                                                 ...    ...  
485  How likely is it that safe, effective vaccines...      1  
486  How quickly could COVID-19 vaccines stop the p...      1  
487  What types of COVID-19 vaccines are being deve...      1  
488  Will other vaccines help protect me from COVID...      1  
489  Will COVID-19 vaccines provide long-term prote...      1  

[5022 rows x 4 columns]

In [629]:
N_Real_Edge[['news_id','tweet_id']]

news_id             tweet_id
0             0  1253800268300455936
1             0  1253800268300455936
2             0  1253800268300455936
3             0  1253800268300455936
4             0  1253800268300455936
...         ...                  ...
115111     4311  1310630581349253124
115112     4311  1310630581349253124
115113     4311  1310630581349253124
115114     4311  1310630581349253124
115115     4311  1319572296319774720

[115116 rows x 2 columns]

In [637]:
C_Real_Edge

news_id             tweet_id             reply_id
0       100000  1253283636843089920  1253282699688771584
1       100000  1253283636843089920  1253285820485767169
2       100001  1253959004436467713  1253791593179611141
3       100001  1253959004436467713  1253802862372704261
4       100001  1253959004436467713  1253811649498230784
...        ...                  ...                  ...
12653   100367  1310967935561728002  1310973543375491072
12654   100367  1310967935561728002  1310975265141530624
12655   100367  1310967935561728002  1310975338487246848
12656   100367  1310967935561728002  1310987368871075842
12657   100338  1310823421014573056  1311308030751932416

[12658 rows x 3 columns]

In [639]:
RealEdge = pd.concat([N_Real_Edge , C_Real_Edge],axis=0)

In [640]:
RealEdge

news_id             tweet_id             reply_id
0            0  1253800268300455936  1253801366327300098
1            0  1253800268300455936  1253801454126665729
2            0  1253800268300455936  1253801598985396224
3            0  1253800268300455936  1253801654882893824
4            0  1253800268300455936  1253802430967341057
...        ...                  ...                  ...
12653   100367  1310967935561728002  1310973543375491072
12654   100367  1310967935561728002  1310975265141530624
12655   100367  1310967935561728002  1310975338487246848
12656   100367  1310967935561728002  1310987368871075842
12657   100338  1310823421014573056  1311308030751932416

[127774 rows x 3 columns]

In [641]:
pd.merge(Real,RealEdge,how='inner')

news_id  fact_check_url  \
0             0  healthline.com   
1             0  healthline.com   
2             0  healthline.com   
3             0  healthline.com   
4             0  healthline.com   
...         ...             ...   
128579   100437         who.int   
128580   100437         who.int   
128581   100445         who.int   
128582   100445         who.int   
128583   100452         who.int   

                                                    title  label  \
0                      Coronavirus Outbreak: Live Updates      1   
1                      Coronavirus Outbreak: Live Updates      1   
2                      Coronavirus Outbreak: Live Updates      1   
3                      Coronavirus Outbreak: Live Updates      1   
4                      Coronavirus Outbreak: Live Updates      1   
...                                                   ...    ...   
128579                               What is ventilation?      1   
128580                               What is ventilation?      1   
128581            Thermal scanners CANNOT detect COVID-19      1   
128582            Thermal scanners CANNOT detect COVID-19      1   
128583  Studies show hydroxychloroquine does not have ...      1   

                   tweet_id             reply_id  
0       1253800268300455936  1253801366327300098  
1       1253800268300455936  1253801454126665729  
2       1253800268300455936  1253801598985396224  
3       1253800268300455936  1253801654882893824  
4       1253800268300455936  1253802430967341057  
...                     ...                  ...  
128579  1282381832168845312  1282382335506354178  
128580  1282381832168845312  1282386486638387207  
128581  1291701648289660928  1291701660054691844  
128582  1286992485487714304  1286992492844507144  
128583  1289225423910367233  1289296900932345856  

[128584 rows x 6 columns]

In [657]:
Real

news_id  fact_check_url  \
0          0  healthline.com   
1          1  healthline.com   
2          2  healthline.com   
3          3  healthline.com   
4          4  healthline.com   
..       ...             ...   
485   100369         who.int   
486   100370         who.int   
487   100371         who.int   
488   100372         who.int   
489   100373         who.int   

                                                 title  label  
0                   Coronavirus Outbreak: Live Updates      1  
1    Screen Time Doesn’t Hurt Kids’ Social Skills, ...      1  
2    Here’s Exactly Where We Are with Vaccines and ...      1  
3    Here’s What Happens to the Body After Contract...      1  
4    Why You Should Be Skeptical of Early Results o...      1  
..                                                 ...    ...  
485  How likely is it that safe, effective vaccines...      1  
486  How quickly could COVID-19 vaccines stop the p...      1  
487  What types of COVID-19 vaccines are being deve...      1  
488  Will other vaccines help protect me from COVID...      1  
489  Will COVID-19 vaccines provide long-term prote...      1  

[5022 rows x 4 columns]

In [658]:
RealEdge

news_id             tweet_id             reply_id
0            0  1253800268300455936  1253801366327300098
1            0  1253800268300455936  1253801454126665729
2            0  1253800268300455936  1253801598985396224
3            0  1253800268300455936  1253801654882893824
4            0  1253800268300455936  1253802430967341057
...        ...                  ...                  ...
12653   100367  1310967935561728002  1310973543375491072
12654   100367  1310967935561728002  1310975265141530624
12655   100367  1310967935561728002  1310975338487246848
12656   100367  1310967935561728002  1310987368871075842
12657   100338  1310823421014573056  1311308030751932416

[127774 rows x 3 columns]

In [666]:
RealNodeFeature = pd.merge(RealEdge,Real,how='inner',on='news_id')[['fact_check_url','title','label','tweet_id']]

In [676]:
RealEdge

news_id             tweet_id             reply_id
0            0  1253800268300455936  1253801366327300098
1            0  1253800268300455936  1253801454126665729
2            0  1253800268300455936  1253801598985396224
3            0  1253800268300455936  1253801654882893824
4            0  1253800268300455936  1253802430967341057
...        ...                  ...                  ...
12653   100367  1310967935561728002  1310973543375491072
12654   100367  1310967935561728002  1310975265141530624
12655   100367  1310967935561728002  1310975338487246848
12656   100367  1310967935561728002  1310987368871075842
12657   100338  1310823421014573056  1311308030751932416

[127774 rows x 3 columns]

### Fake

In [648]:
FakeEdge = pd.concat([C_Fake_Edge , N_Fake_Edge],axis=0)

In [651]:
FakeNodeFeature = pd.merge(Fake,FakeEdge,how='inner')[['tweet_id','fact_check_url','title','label']]

## Node Feature result

In [652]:
RealNodeFeature

fact_check_url                                              title  \
0       healthline.com                 Coronavirus Outbreak: Live Updates   
1       healthline.com                 Coronavirus Outbreak: Live Updates   
2       healthline.com                 Coronavirus Outbreak: Live Updates   
3       healthline.com                 Coronavirus Outbreak: Live Updates   
4       healthline.com                 Coronavirus Outbreak: Live Updates   
...                ...                                                ...   
128579         who.int                               What is ventilation?   
128580         who.int                               What is ventilation?   
128581         who.int            Thermal scanners CANNOT detect COVID-19   
128582         who.int            Thermal scanners CANNOT detect COVID-19   
128583         who.int  Studies show hydroxychloroquine does not have ...   

        label             tweet_id  
0           1  1253800268300455936  
1           1  1253800268300455936  
2           1  1253800268300455936  
3           1  1253800268300455936  
4           1  1253800268300455936  
...       ...                  ...  
128579      1  1282381832168845312  
128580      1  1282381832168845312  
128581      1  1291701648289660928  
128582      1  1286992485487714304  
128583      1  1289225423910367233  

[128584 rows x 4 columns]

In [653]:
FakeNodeFeature

tweet_id        fact_check_url  \
0     1255484783884369921    healthfeedback.org   
1     1255484783884369921    healthfeedback.org   
2     1255484783884369921    healthfeedback.org   
3     1249534887691984897    healthfeedback.org   
4     1249534887691984897    healthfeedback.org   
...                   ...                   ...   
8098  1240247344131493888  medicalnewstoday.com   
8099  1240247344131493888  medicalnewstoday.com   
8100  1271261017918472192  medicalnewstoday.com   
8101  1269957159724265472  medicalnewstoday.com   
8102  1269957159724265472  medicalnewstoday.com   

                                                  title  label  
0     Nobel laureate Luc Montagnier claimed that the...      0  
1     Nobel laureate Luc Montagnier claimed that the...      0  
2     Nobel laureate Luc Montagnier claimed that the...      0  
3     A video argues that the pandemic could have re...      0  
4     A video argues that the pandemic could have re...      0  
...                                                 ...    ...  
8098    "The virus originated in a laboratory in China"      0  
8099    "The virus originated in a laboratory in China"      0  
8100        You can catch coronavirus in swimming pools      0  
8101        You can catch coronavirus in swimming pools      0  
8102        You can catch coronavirus in swimming pools      0  

[8103 rows x 4 columns]

In [655]:
RealEdge

news_id             tweet_id             reply_id
0            0  1253800268300455936  1253801366327300098
1            0  1253800268300455936  1253801454126665729
2            0  1253800268300455936  1253801598985396224
3            0  1253800268300455936  1253801654882893824
4            0  1253800268300455936  1253802430967341057
...        ...                  ...                  ...
12653   100367  1310967935561728002  1310973543375491072
12654   100367  1310967935561728002  1310975265141530624
12655   100367  1310967935561728002  1310975338487246848
12656   100367  1310967935561728002  1310987368871075842
12657   100338  1310823421014573056  1311308030751932416

[127774 rows x 3 columns]

In [654]:
FakeEdge

news_id             tweet_id             reply_id
0      100001  1242474839966765056  1242474835512446976
1      100001  1242474839966765056  1242474843137703938
2      100001  1242400142335983619  1242388627943878657
3      100001  1242400142335983619  1242400453700096001
4      100003  1248606541793488896  1248609022841729028
...       ...                  ...                  ...
7472      968  1306185322540171265  1306219600040861696
7473      968  1306185322540171265  1306221112980725760
7474      968  1306185322540171265  1306222654597152768
7475      968  1306265744062255104  1306268765848588297
7476      968  1306265744062255104  1306269570378932224

[8103 rows x 3 columns]

## Sentence Embedding

In [692]:
RealNodeFeature

fact_check_url                                              title  \
0       healthline.com                 Coronavirus Outbreak: Live Updates   
1       healthline.com                 Coronavirus Outbreak: Live Updates   
2       healthline.com                 Coronavirus Outbreak: Live Updates   
3       healthline.com                 Coronavirus Outbreak: Live Updates   
4       healthline.com                 Coronavirus Outbreak: Live Updates   
...                ...                                                ...   
128579         who.int                   Is there a vaccine for COVID-19?   
128580         who.int  Are there any herbal teas or herbal supplement...   
128581         who.int                   Is there a vaccine for COVID-19?   
128582         who.int  Are there any herbal teas or herbal supplement...   
128583         who.int                   Is there a vaccine for COVID-19?   

        label             tweet_id  
0           1  1253800268300455936  
1           1  1253800268300455936  
2           1  1253800268300455936  
3           1  1253800268300455936  
4           1  1253800268300455936  
...       ...                  ...  
128579      1  1310967935561728002  
128580      1  1310967935561728002  
128581      1  1310967935561728002  
128582      1  1310967935561728002  
128583      1  1310967935561728002  

[128584 rows x 4 columns]

In [693]:
FakeNodeFeature

tweet_id        fact_check_url  \
0     1255484783884369921    healthfeedback.org   
1     1255484783884369921    healthfeedback.org   
2     1255484783884369921    healthfeedback.org   
3     1249534887691984897    healthfeedback.org   
4     1249534887691984897    healthfeedback.org   
...                   ...                   ...   
8098  1240247344131493888  medicalnewstoday.com   
8099  1240247344131493888  medicalnewstoday.com   
8100  1271261017918472192  medicalnewstoday.com   
8101  1269957159724265472  medicalnewstoday.com   
8102  1269957159724265472  medicalnewstoday.com   

                                                  title  label  
0     Nobel laureate Luc Montagnier claimed that the...      0  
1     Nobel laureate Luc Montagnier claimed that the...      0  
2     Nobel laureate Luc Montagnier claimed that the...      0  
3     A video argues that the pandemic could have re...      0  
4     A video argues that the pandemic could have re...      0  
...                                                 ...    ...  
8098    "The virus originated in a laboratory in China"      0  
8099    "The virus originated in a laboratory in China"      0  
8100        You can catch coronavirus in swimming pools      0  
8101        You can catch coronavirus in swimming pools      0  
8102        You can catch coronavirus in swimming pools      0  

[8103 rows x 4 columns]

In [1]:
!ls

0628.ipynb	  FakeNodeFeature.csv	    Real.csv
0629_trial.ipynb  FeatureEngeneering.ipynb  RealEdge.csv
Baseline.ipynb	  GCN.ipynb		    RealNodeFeature.csv
CoAID		  GraphSAGE.ipynb	    test_dgl.ipynb
dataset		  logs			    test_stellargraph.ipynb
dataset2	  model.pt		    Untitled1.ipynb
Fake.csv	  newdataset		    Untitled2.ipynb
FakeEdge.csv	  products.zip		    Untitled.ipynb


In [2]:
from sentence_transformers import SentenceTransformer
model = SentenceTransformer('paraphrase-distilroberta-base-v2')

Downloading:   0%|          | 0.00/736 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/3.72k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/686 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/122 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/456k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/229 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/329M [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/53.0 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/239 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/1.12k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/798k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/190 [00:00<?, ?B/s]

In [5]:
RealNodeFeature = pd.read_csv('RealNodeFeature.csv')
FakeNodeFeature = pd.read_csv('FakeNodeFeature.csv')
RealEdge = pd.read_csv('RealEdge.csv')
FakeEdge = pd.read_csv('FakeEdge.csv')

# Feature Engineering

In [9]:
RealTitleEncoding = model.encode(RealNodeFeature['title'])
col_name = range(len(RealTitleEncoding[0]))
for n in col_name:
    RealNodeFeature[n] = [m[n] for m in RealTitleEncoding]

In [13]:
del RealNodeFeature['Unnamed: 0']

In [15]:
RealNodeFeature

fact_check_url                                              title  \
0       healthline.com                 Coronavirus Outbreak: Live Updates   
1       healthline.com                 Coronavirus Outbreak: Live Updates   
2       healthline.com                 Coronavirus Outbreak: Live Updates   
3       healthline.com                 Coronavirus Outbreak: Live Updates   
4       healthline.com                 Coronavirus Outbreak: Live Updates   
...                ...                                                ...   
128579         who.int                   Is there a vaccine for COVID-19?   
128580         who.int  Are there any herbal teas or herbal supplement...   
128581         who.int                   Is there a vaccine for COVID-19?   
128582         who.int  Are there any herbal teas or herbal supplement...   
128583         who.int                   Is there a vaccine for COVID-19?   

        label             tweet_id         0         1         2         3  \
0           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
1           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
2           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
3           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
4           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
...       ...                  ...       ...       ...       ...       ...   
128579      1  1310967935561728002  0.022426 -0.045059 -0.556697 -1.069443   
128580      1  1310967935561728002  0.688832  0.630236 -0.025951 -0.937067   
128581      1  1310967935561728002  0.022426 -0.045059 -0.556697 -1.069443   
128582      1  1310967935561728002  0.688832  0.630236 -0.025951 -0.937067   
128583      1  1310967935561728002  0.022426 -0.045059 -0.556697 -1.069443   

               4         5  ...       758       759       760       761  \
0      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
1      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
2      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
3      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
4      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
...          ...       ...  ...       ...       ...       ...       ...   
128579 -0.681256  0.900455  ...  0.274253 -0.591422 -0.187397 -0.330570   
128580  0.062282  0.628631  ... -0.344029 -1.251898  0.281046  0.014782   
128581 -0.681256  0.900455  ...  0.274253 -0.591422 -0.187397 -0.330570   
128582  0.062282  0.628631  ... -0.344029 -1.251898  0.281046  0.014782   
128583 -0.681256  0.900455  ...  0.274253 -0.591422 -0.187397 -0.330570   

             762       763       764       765       766       767  
0       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
1       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
2       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
3       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
4       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
...          ...       ...       ...       ...       ...       ...  
128579  0.901808 -0.402947 -0.306542  2.362385 -0.142832 -0.486116  
128580 -0.050562  0.078907 -0.562974  1.573720  0.171875 -0.257370  
128581  0.901808 -0.402947 -0.306542  2.362385 -0.142832 -0.486116  
128582 -0.050562  0.078907 -0.562974  1.573720  0.171875 -0.257370  
128583  0.901808 -0.402947 -0.306542  2.362385 -0.142832 -0.486116  

[128584 rows x 772 columns]

In [16]:
FakeTitleEncoding = model.encode(FakeNodeFeature['title'])
col_name = range(len(FakeTitleEncoding[0]))
for n in col_name:
    FakeNodeFeature[n] = [m[n] for m in FakeTitleEncoding]

In [ ]:
del FakeNodeFaeture['Unnamed: 0']

In [18]:
FakeNodeFeature

tweet_id        fact_check_url  \
0     1255484783884369921    healthfeedback.org   
1     1255484783884369921    healthfeedback.org   
2     1255484783884369921    healthfeedback.org   
3     1249534887691984897    healthfeedback.org   
4     1249534887691984897    healthfeedback.org   
...                   ...                   ...   
8098  1240247344131493888  medicalnewstoday.com   
8099  1240247344131493888  medicalnewstoday.com   
8100  1271261017918472192  medicalnewstoday.com   
8101  1269957159724265472  medicalnewstoday.com   
8102  1269957159724265472  medicalnewstoday.com   

                                                  title  label         0  \
0     Nobel laureate Luc Montagnier claimed that the...      0 -0.785891   
1     Nobel laureate Luc Montagnier claimed that the...      0 -0.785891   
2     Nobel laureate Luc Montagnier claimed that the...      0 -0.785891   
3     A video argues that the pandemic could have re...      0 -0.328232   
4     A video argues that the pandemic could have re...      0 -0.328232   
...                                                 ...    ...       ...   
8098    "The virus originated in a laboratory in China"      0  0.034284   
8099    "The virus originated in a laboratory in China"      0  0.034284   
8100        You can catch coronavirus in swimming pools      0  0.818228   
8101        You can catch coronavirus in swimming pools      0  0.818228   
8102        You can catch coronavirus in swimming pools      0  0.818228   

             1         2         3         4         5  ...       758  \
0    -0.161625 -0.050717 -0.320772 -0.802968  0.731735  ...  0.207789   
1    -0.161625 -0.050717 -0.320772 -0.802968  0.731735  ...  0.207789   
2    -0.161625 -0.050717 -0.320772 -0.802968  0.731735  ...  0.207789   
3    -0.183012 -0.328267 -0.049121  0.145228  1.374485  ...  0.288235   
4    -0.183012 -0.328267 -0.049121  0.145228  1.374485  ...  0.288235   
...        ...       ...       ...       ...       ...  ...       ...   
8098 -0.945763 -0.254773 -0.051217  0.069432  0.803254  ...  0.991776   
8099 -0.945763 -0.254773 -0.051217  0.069432  0.803254  ...  0.991776   
8100  0.185544  0.084825 -0.824511 -0.313027  0.762159  ...  0.992290   
8101  0.185544  0.084825 -0.824511 -0.313027  0.762159  ...  0.992290   
8102  0.185543  0.084825 -0.824511 -0.313027  0.762159  ...  0.992290   

           759       760       761       762       763       764       765  \
0    -0.326870 -0.173945  0.074450  0.678072  0.182988 -0.200717  0.657090   
1    -0.326870 -0.173945  0.074450  0.678072  0.182988 -0.200717  0.657090   
2    -0.326870 -0.173945  0.074450  0.678072  0.182988 -0.200717  0.657090   
3    -0.021198 -0.520571 -0.321243  0.744026  0.136966  0.303735 -0.248444   
4    -0.021198 -0.520571 -0.321243  0.744026  0.136966  0.303735 -0.248444   
...        ...       ...       ...       ...       ...       ...       ...   
8098 -0.542556 -0.979326 -0.449397  0.312563  0.278845  0.676542 -0.222831   
8099 -0.542556 -0.979326 -0.449397  0.312563  0.278845  0.676542 -0.222831   
8100 -0.868523 -0.897630 -0.716556  1.150071  0.116452 -0.783419  0.844500   
8101 -0.868523 -0.897630 -0.716556  1.150071  0.116452 -0.783419  0.844500   
8102 -0.868523 -0.897630 -0.716556  1.150071  0.116452 -0.783419  0.844499   

           766       767  
0    -0.473048 -0.101156  
1    -0.473048 -0.101156  
2    -0.473048 -0.101156  
3    -0.280551 -0.090947  
4    -0.280551 -0.090947  
...        ...       ...  
8098 -0.266015 -0.557793  
8099 -0.266015 -0.557793  
8100 -0.255630  0.239454  
8101 -0.255630  0.239454  
8102 -0.255630  0.239455  

[8103 rows x 772 columns]

In [21]:
RealNodeFeature['fact_check_url']

0         healthline.com
1         healthline.com
2         healthline.com
3         healthline.com
4         healthline.com
               ...      
128579           who.int
128580           who.int
128581           who.int
128582           who.int
128583           who.int
Name: fact_check_url, Length: 128584, dtype: object

In [24]:
RealNodeFeature['fact_check_url'].value_counts()

cdc.gov                 92345
who.int                 13177
webmd.com               12275
medicalnewstoday.com     3037
sciencedaily.com         2759
twitter.com              1787
nih.gov                  1613
healthline.com            936
clevelandclinic.org       519
mayoclinic.org             72
politifact.com             64
Name: fact_check_url, dtype: int64

In [22]:
FakeNodeFeature['fact_check_url']

0         healthfeedback.org
1         healthfeedback.org
2         healthfeedback.org
3         healthfeedback.org
4         healthfeedback.org
                ...         
8098    medicalnewstoday.com
8099    medicalnewstoday.com
8100    medicalnewstoday.com
8101    medicalnewstoday.com
8102    medicalnewstoday.com
Name: fact_check_url, Length: 8103, dtype: object

In [25]:
FakeNodeFeature['fact_check_url'].value_counts()

www.politifact.com            2761
leadstories.com               2071
hoax-alert.leadstories.com    1257
healthfeedback.org             761
medicalnewstoday.com           626
www.factcheck.org              606
www.washingtonpost.com          20
factcheck.afp.com                1
Name: fact_check_url, dtype: int64

In [33]:
RealNodeFeature

fact_check_url                                              title  \
0       healthline.com                 Coronavirus Outbreak: Live Updates   
1       healthline.com                 Coronavirus Outbreak: Live Updates   
2       healthline.com                 Coronavirus Outbreak: Live Updates   
3       healthline.com                 Coronavirus Outbreak: Live Updates   
4       healthline.com                 Coronavirus Outbreak: Live Updates   
...                ...                                                ...   
128579         who.int                   Is there a vaccine for COVID-19?   
128580         who.int  Are there any herbal teas or herbal supplement...   
128581         who.int                   Is there a vaccine for COVID-19?   
128582         who.int  Are there any herbal teas or herbal supplement...   
128583         who.int                   Is there a vaccine for COVID-19?   

        label             tweet_id         0         1         2         3  \
0           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
1           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
2           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
3           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
4           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
...       ...                  ...       ...       ...       ...       ...   
128579      1  1310967935561728002  0.022426 -0.045059 -0.556697 -1.069443   
128580      1  1310967935561728002  0.688832  0.630236 -0.025951 -0.937067   
128581      1  1310967935561728002  0.022426 -0.045059 -0.556697 -1.069443   
128582      1  1310967935561728002  0.688832  0.630236 -0.025951 -0.937067   
128583      1  1310967935561728002  0.022426 -0.045059 -0.556697 -1.069443   

               4         5  ...       758       759       760       761  \
0      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
1      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
2      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
3      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
4      -0.592689  0.917206  ...  0.223209 -0.586760 -0.449738 -0.418782   
...          ...       ...  ...       ...       ...       ...       ...   
128579 -0.681256  0.900455  ...  0.274253 -0.591422 -0.187397 -0.330570   
128580  0.062282  0.628631  ... -0.344029 -1.251898  0.281046  0.014782   
128581 -0.681256  0.900455  ...  0.274253 -0.591422 -0.187397 -0.330570   
128582  0.062282  0.628631  ... -0.344029 -1.251898  0.281046  0.014782   
128583 -0.681256  0.900455  ...  0.274253 -0.591422 -0.187397 -0.330570   

             762       763       764       765       766       767  
0       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
1       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
2       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
3       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
4       1.267020  0.189194 -0.257334  1.044064 -0.386478  0.166367  
...          ...       ...       ...       ...       ...       ...  
128579  0.901808 -0.402947 -0.306542  2.362385 -0.142832 -0.486116  
128580 -0.050562  0.078907 -0.562974  1.573720  0.171875 -0.257370  
128581  0.901808 -0.402947 -0.306542  2.362385 -0.142832 -0.486116  
128582 -0.050562  0.078907 -0.562974  1.573720  0.171875 -0.257370  
128583  0.901808 -0.402947 -0.306542  2.362385 -0.142832 -0.486116  

[128584 rows x 772 columns]

In [59]:
# column 재배치
col1 = FakeNodeFeature[['fact_check_url','title','label','tweet_id']].columns.to_list()
col2 = FakeNodeFeature.columns[4:].to_list()
new_col = col1+col2
FakeNodeFeature = FakeNodeFeature[new_col]

In [68]:
FakeNodeFeature

fact_check_url                                              title  \
0       healthfeedback.org  Nobel laureate Luc Montagnier claimed that the...   
1       healthfeedback.org  Nobel laureate Luc Montagnier claimed that the...   
2       healthfeedback.org  Nobel laureate Luc Montagnier claimed that the...   
3       healthfeedback.org  A video argues that the pandemic could have re...   
4       healthfeedback.org  A video argues that the pandemic could have re...   
...                    ...                                                ...   
8098  medicalnewstoday.com    "The virus originated in a laboratory in China"   
8099  medicalnewstoday.com    "The virus originated in a laboratory in China"   
8100  medicalnewstoday.com        You can catch coronavirus in swimming pools   
8101  medicalnewstoday.com        You can catch coronavirus in swimming pools   
8102  medicalnewstoday.com        You can catch coronavirus in swimming pools   

      label             tweet_id         0         1         2         3  \
0         0  1255484783884369921 -0.785891 -0.161625 -0.050717 -0.320772   
1         0  1255484783884369921 -0.785891 -0.161625 -0.050717 -0.320772   
2         0  1255484783884369921 -0.785891 -0.161625 -0.050717 -0.320772   
3         0  1249534887691984897 -0.328232 -0.183012 -0.328267 -0.049121   
4         0  1249534887691984897 -0.328232 -0.183012 -0.328267 -0.049121   
...     ...                  ...       ...       ...       ...       ...   
8098      0  1240247344131493888  0.034284 -0.945763 -0.254773 -0.051217   
8099      0  1240247344131493888  0.034284 -0.945763 -0.254773 -0.051217   
8100      0  1271261017918472192  0.818228  0.185544  0.084825 -0.824511   
8101      0  1269957159724265472  0.818228  0.185544  0.084825 -0.824511   
8102      0  1269957159724265472  0.818228  0.185543  0.084825 -0.824511   

             4         5  ...       758       759       760       761  \
0    -0.802968  0.731735  ...  0.207789 -0.326870 -0.173945  0.074450   
1    -0.802968  0.731735  ...  0.207789 -0.326870 -0.173945  0.074450   
2    -0.802968  0.731735  ...  0.207789 -0.326870 -0.173945  0.074450   
3     0.145228  1.374485  ...  0.288235 -0.021198 -0.520571 -0.321243   
4     0.145228  1.374485  ...  0.288235 -0.021198 -0.520571 -0.321243   
...        ...       ...  ...       ...       ...       ...       ...   
8098  0.069432  0.803254  ...  0.991776 -0.542556 -0.979326 -0.449397   
8099  0.069432  0.803254  ...  0.991776 -0.542556 -0.979326 -0.449397   
8100 -0.313027  0.762159  ...  0.992290 -0.868523 -0.897630 -0.716556   
8101 -0.313027  0.762159  ...  0.992290 -0.868523 -0.897630 -0.716556   
8102 -0.313027  0.762159  ...  0.992290 -0.868523 -0.897630 -0.716556   

           762       763       764       765       766       767  
0     0.678072  0.182988 -0.200717  0.657090 -0.473048 -0.101156  
1     0.678072  0.182988 -0.200717  0.657090 -0.473048 -0.101156  
2     0.678072  0.182988 -0.200717  0.657090 -0.473048 -0.101156  
3     0.744026  0.136966  0.303735 -0.248444 -0.280551 -0.090947  
4     0.744026  0.136966  0.303735 -0.248444 -0.280551 -0.090947  
...        ...       ...       ...       ...       ...       ...  
8098  0.312563  0.278845  0.676542 -0.222831 -0.266015 -0.557793  
8099  0.312563  0.278845  0.676542 -0.222831 -0.266015 -0.557793  
8100  1.150071  0.116452 -0.783419  0.844500 -0.255630  0.239454  
8101  1.150071  0.116452 -0.783419  0.844500 -0.255630  0.239454  
8102  1.150071  0.116452 -0.783419  0.844499 -0.255630  0.239455  

[8103 rows x 772 columns]

In [107]:
# Real node , Fake node category feature 종합을 위함.
NodeFeature = pd.concat([RealNodeFeature, FakeNodeFeature],axis=0)
# Agency 추출 하기위한 dummies
Agency = pd.get_dummies(NodeFeature['fact_check_url'])
# 추출된 데이터를 토대로 axis=1 로 붙여줌.
NodeFeature = pd.concat([NodeFeature, Agency],axis=1)
NodeFinal = NodeFeature.copy()

In [111]:
del NodeFinal['fact_check_url']
del NodeFinal['title']

In [217]:
NodeFinal['label'].value_counts()

1    128584
0      8103
Name: label, dtype: int64

In [90]:
Real_Edge = RealEdge[['tweet_id', 'reply_id']]
Fake_Edge = FakeEdge[['tweet_id', 'reply_id']]

## Undersampling

In [210]:
node_features

tweet_id         0         1         2         3         4  \
95737   1265934298688913408 -0.116209  0.167043  0.056090  0.137774  0.163017   
85117   1278093396469403649  0.203299  0.772909  0.263678  0.151804 -0.360949   
118672  1240948885910179841 -0.421290  0.236152 -0.176450 -0.137431 -0.166321   
90860   1277318156554579972  0.238420  0.083957 -0.225509  0.233015  0.900433   
17371   1255510821406478337 -0.321673  0.377406 -0.779668  0.105812  0.579647   
...                     ...       ...       ...       ...       ...       ...   
4628    1225690115206287361 -0.474417 -0.029161 -0.421501  0.199111 -0.804776   
3320    1233897920564465664 -0.281977  0.205518  0.672352 -0.408573 -1.152401   
2578    1253365473309270016 -0.111937  0.091093  0.022958 -0.732366 -1.159551   
4571    1224101000396578819 -0.567308 -0.128500  0.208987  0.274314 -0.697975   
3454    1233840642809311239 -0.281977  0.205518  0.672352 -0.408573 -1.152401   

               5         6         7         8  ...  medicalnewstoday.com  \
95737  -0.013151  0.060183  0.298546  0.518897  ...                     0   
85117   0.686531 -0.140601  0.051136  0.024650  ...                     0   
118672 -0.669668  0.353261 -0.505883 -0.799428  ...                     0   
90860   1.030108 -0.322384 -0.671392 -0.796842  ...                     0   
17371   0.826184 -0.478850  0.439447 -0.517407  ...                     0   
...          ...       ...       ...       ...  ...                   ...   
4628    0.578414  0.019149  0.376171  0.006558  ...                     0   
3320    0.392728  0.376377  0.192601 -0.742051  ...                     0   
2578   -0.062528  0.327891  0.219396 -0.926162  ...                     0   
4571    0.789915  0.089886  0.885885 -0.247533  ...                     0   
3454    0.392728  0.376377  0.192601 -0.742051  ...                     0   

        nih.gov  politifact.com  sciencedaily.com  twitter.com  webmd.com  \
95737         0               0                 0            0          0   
85117         0               0                 0            0          0   
118672        0               0                 0            0          0   
90860         0               0                 0            0          0   
17371         0               0                 0            0          0   
...         ...             ...               ...          ...        ...   
4628          0               0                 0            0          0   
3320          0               0                 0            0          0   
2578          0               0                 0            0          0   
4571          0               0                 0            0          0   
3454          0               0                 0            0          0   

        who.int  www.factcheck.org  www.politifact.com  www.washingtonpost.com  
95737         0                  0                   0                       0  
85117         0                  0                   0                       0  
118672        1                  0                   0                       0  
90860         0                  0                   0                       0  
17371         0                  0                   0                       0  
...         ...                ...                 ...                     ...  
4628          0                  0                   1                       0  
3320          0                  0                   0                       0  
2578          0                  0                   1                       0  
4571          0                  0                   0                       0  
3454          0                  0                   0                       0  

[16206 rows x 787 columns]

In [138]:
NodeFinalReal = NodeFinal[NodeFinal['label'] == 1].sample(NodeFinalFake.shape[0])
NodeFinalFake = NodeFinal[NodeFinal['label'] == 0]

In [116]:
node_features = pd.concat([NodeFinalReal , NodeFinalFake],axis=0)

In [206]:
node_features

tweet_id         0         1         2         3         4  \
95737   1265934298688913408 -0.116209  0.167043  0.056090  0.137774  0.163017   
85117   1278093396469403649  0.203299  0.772909  0.263678  0.151804 -0.360949   
118672  1240948885910179841 -0.421290  0.236152 -0.176450 -0.137431 -0.166321   
90860   1277318156554579972  0.238420  0.083957 -0.225509  0.233015  0.900433   
17371   1255510821406478337 -0.321673  0.377406 -0.779668  0.105812  0.579647   
...                     ...       ...       ...       ...       ...       ...   
4628    1225690115206287361 -0.474417 -0.029161 -0.421501  0.199111 -0.804776   
3320    1233897920564465664 -0.281977  0.205518  0.672352 -0.408573 -1.152401   
2578    1253365473309270016 -0.111937  0.091093  0.022958 -0.732366 -1.159551   
4571    1224101000396578819 -0.567308 -0.128500  0.208987  0.274314 -0.697975   
3454    1233840642809311239 -0.281977  0.205518  0.672352 -0.408573 -1.152401   

               5         6         7         8  ...  medicalnewstoday.com  \
95737  -0.013151  0.060183  0.298546  0.518897  ...                     0   
85117   0.686531 -0.140601  0.051136  0.024650  ...                     0   
118672 -0.669668  0.353261 -0.505883 -0.799428  ...                     0   
90860   1.030108 -0.322384 -0.671392 -0.796842  ...                     0   
17371   0.826184 -0.478850  0.439447 -0.517407  ...                     0   
...          ...       ...       ...       ...  ...                   ...   
4628    0.578414  0.019149  0.376171  0.006558  ...                     0   
3320    0.392728  0.376377  0.192601 -0.742051  ...                     0   
2578   -0.062528  0.327891  0.219396 -0.926162  ...                     0   
4571    0.789915  0.089886  0.885885 -0.247533  ...                     0   
3454    0.392728  0.376377  0.192601 -0.742051  ...                     0   

        nih.gov  politifact.com  sciencedaily.com  twitter.com  webmd.com  \
95737         0               0                 0            0          0   
85117         0               0                 0            0          0   
118672        0               0                 0            0          0   
90860         0               0                 0            0          0   
17371         0               0                 0            0          0   
...         ...             ...               ...          ...        ...   
4628          0               0                 0            0          0   
3320          0               0                 0            0          0   
2578          0               0                 0            0          0   
4571          0               0                 0            0          0   
3454          0               0                 0            0          0   

        who.int  www.factcheck.org  www.politifact.com  www.washingtonpost.com  
95737         0                  0                   0                       0  
85117         0                  0                   0                       0  
118672        1                  0                   0                       0  
90860         0                  0                   0                       0  
17371         0                  0                   0                       0  
...         ...                ...                 ...                     ...  
4628          0                  0                   1                       0  
3320          0                  0                   0                       0  
2578          0                  0                   1                       0  
4571          0                  0                   0                       0  
3454          0                  0                   0                       0  

[16206 rows x 787 columns]

In [118]:
node_labels = node_features.pop('label')

In [205]:
node_labels

95737     1
85117     1
118672    1
90860     1
17371     1
         ..
4628      0
3320      0
2578      0
4571      0
3454      0
Name: label, Length: 16206, dtype: int64

## Edge Engineering

In [192]:
RealEdgeSample = Real_Edge.sample(8103)

In [122]:
Fake_Edge

tweet_id             reply_id
0     1242474839966765056  1242474835512446976
1     1242474839966765056  1242474843137703938
2     1242400142335983619  1242388627943878657
3     1242400142335983619  1242400453700096001
4     1248606541793488896  1248609022841729028
...                   ...                  ...
8098  1306185322540171265  1306219600040861696
8099  1306185322540171265  1306221112980725760
8100  1306185322540171265  1306222654597152768
8101  1306265744062255104  1306268765848588297
8102  1306265744062255104  1306269570378932224

[8103 rows x 2 columns]

In [119]:
edges = pd.concat([Real_Edge, Fake_Edge],axis=0)

In [120]:
edges

tweet_id             reply_id
0     1253800268300455936  1253801366327300098
1     1253800268300455936  1253801454126665729
2     1253800268300455936  1253801598985396224
3     1253800268300455936  1253801654882893824
4     1253800268300455936  1253802430967341057
...                   ...                  ...
8098  1306185322540171265  1306219600040861696
8099  1306185322540171265  1306221112980725760
8100  1306185322540171265  1306222654597152768
8101  1306265744062255104  1306268765848588297
8102  1306265744062255104  1306269570378932224

[135877 rows x 2 columns]

## just go

In [249]:
Test = NodeFinal[NodeFinal['label'] == 1]

In [228]:
Real_Edge

tweet_id             reply_id
0       1253800268300455936  1253801366327300098
1       1253800268300455936  1253801454126665729
2       1253800268300455936  1253801598985396224
3       1253800268300455936  1253801654882893824
4       1253800268300455936  1253802430967341057
...                     ...                  ...
127769  1310967935561728002  1310973543375491072
127770  1310967935561728002  1310975265141530624
127771  1310967935561728002  1310975338487246848
127772  1310967935561728002  1310987368871075842
127773  1310823421014573056  1311308030751932416

[127774 rows x 2 columns]

## 데이터 프레임 비교를 통한 교집합 가져오기 merge 변형

In [252]:
Test2 = pd.merge(Test, Real_Edge,
                how='outer',
                left_index = True,
                right_index = True,
                indicator = True
                )

In [256]:
Real_Edge[Real_Edge.index.isin(Test2[Test2['_merge'] == 'both'].index)]

tweet_id             reply_id
0       1253800268300455936  1253801366327300098
1       1253800268300455936  1253801454126665729
2       1253800268300455936  1253801598985396224
3       1253800268300455936  1253801654882893824
4       1253800268300455936  1253802430967341057
...                     ...                  ...
127769  1310967935561728002  1310973543375491072
127770  1310967935561728002  1310975265141530624
127771  1310967935561728002  1310975338487246848
127772  1310967935561728002  1310987368871075842
127773  1310823421014573056  1311308030751932416

[127774 rows x 2 columns]

In [257]:
Test[Test.index.isin(Test2[Test2['_merge'] == 'both'].index)]

label             tweet_id         0         1         2         3  \
0           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
1           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
2           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
3           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
4           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
...       ...                  ...       ...       ...       ...       ...   
127769      1  1291114454105505792 -0.345220  0.026889  0.203913  1.021343   
127770      1  1291114454105505792  0.173867 -0.214922 -1.008380  0.528001   
127771      1  1291114454105505792 -0.345220  0.026889  0.203913  1.021343   
127772      1  1291114454105505792  0.173867 -0.214922 -1.008380  0.528001   
127773      1  1291114454105505792 -0.345220  0.026889  0.203913  1.021343   

               4         5         6         7  ...  medicalnewstoday.com  \
0      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
1      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
2      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
3      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
4      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
...          ...       ...       ...       ...  ...                   ...   
127769 -0.525933 -0.294129 -0.673349  0.262235  ...                     0   
127770  0.619187  0.617954  0.113712 -0.230034  ...                     0   
127771 -0.525933 -0.294129 -0.673349  0.262235  ...                     0   
127772  0.619187  0.617954  0.113712 -0.230034  ...                     0   
127773 -0.525933 -0.294129 -0.673349  0.262235  ...                     0   

        nih.gov  politifact.com  sciencedaily.com  twitter.com  webmd.com  \
0             0               0                 0            0          0   
1             0               0                 0            0          0   
2             0               0                 0            0          0   
3             0               0                 0            0          0   
4             0               0                 0            0          0   
...         ...             ...               ...          ...        ...   
127769        0               0                 0            0          0   
127770        0               0                 0            0          0   
127771        0               0                 0            0          0   
127772        0               0                 0            0          0   
127773        0               0                 0            0          0   

        who.int  www.factcheck.org  www.politifact.com  www.washingtonpost.com  
0             0                  0                   0                       0  
1             0                  0                   0                       0  
2             0                  0                   0                       0  
3             0                  0                   0                       0  
4             0                  0                   0                       0  
...         ...                ...                 ...                     ...  
127769        1                  0                   0                       0  
127770        1                  0                   0                       0  
127771        1                  0                   0                       0  
127772        1                  0                   0                       0  
127773        1                  0                   0                       0  

[127774 rows x 788 columns]

In [260]:
Node_Real = Test[Test.index.isin(Test2[Test2['_merge'] == 'both'].index)]

In [264]:
Node_Fake = NodeFinal[NodeFinal['label'] == 0]

In [265]:
Node_Real

label             tweet_id         0         1         2         3  \
0           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
1           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
2           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
3           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
4           1  1253800268300455936 -0.336066 -0.059216  0.047237  0.693768   
...       ...                  ...       ...       ...       ...       ...   
127769      1  1291114454105505792 -0.345220  0.026889  0.203913  1.021343   
127770      1  1291114454105505792  0.173867 -0.214922 -1.008380  0.528001   
127771      1  1291114454105505792 -0.345220  0.026889  0.203913  1.021343   
127772      1  1291114454105505792  0.173867 -0.214922 -1.008380  0.528001   
127773      1  1291114454105505792 -0.345220  0.026889  0.203913  1.021343   

               4         5         6         7  ...  medicalnewstoday.com  \
0      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
1      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
2      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
3      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
4      -0.592689  0.917206 -0.351770 -0.450694  ...                     0   
...          ...       ...       ...       ...  ...                   ...   
127769 -0.525933 -0.294129 -0.673349  0.262235  ...                     0   
127770  0.619187  0.617954  0.113712 -0.230034  ...                     0   
127771 -0.525933 -0.294129 -0.673349  0.262235  ...                     0   
127772  0.619187  0.617954  0.113712 -0.230034  ...                     0   
127773 -0.525933 -0.294129 -0.673349  0.262235  ...                     0   

        nih.gov  politifact.com  sciencedaily.com  twitter.com  webmd.com  \
0             0               0                 0            0          0   
1             0               0                 0            0          0   
2             0               0                 0            0          0   
3             0               0                 0            0          0   
4             0               0                 0            0          0   
...         ...             ...               ...          ...        ...   
127769        0               0                 0            0          0   
127770        0               0                 0            0          0   
127771        0               0                 0            0          0   
127772        0               0                 0            0          0   
127773        0               0                 0            0          0   

        who.int  www.factcheck.org  www.politifact.com  www.washingtonpost.com  
0             0                  0                   0                       0  
1             0                  0                   0                       0  
2             0                  0                   0                       0  
3             0                  0                   0                       0  
4             0                  0                   0                       0  
...         ...                ...                 ...                     ...  
127769        1                  0                   0                       0  
127770        1                  0                   0                       0  
127771        1                  0                   0                       0  
127772        1                  0                   0                       0  
127773        1                  0                   0                       0  

[127774 rows x 788 columns]

In [266]:
Node_Fake

label             tweet_id         0         1         2         3  \
0         0  1255484783884369921 -0.785891 -0.161625 -0.050717 -0.320772   
1         0  1255484783884369921 -0.785891 -0.161625 -0.050717 -0.320772   
2         0  1255484783884369921 -0.785891 -0.161625 -0.050717 -0.320772   
3         0  1249534887691984897 -0.328232 -0.183012 -0.328267 -0.049121   
4         0  1249534887691984897 -0.328232 -0.183012 -0.328267 -0.049121   
...     ...                  ...       ...       ...       ...       ...   
8098      0  1240247344131493888  0.034284 -0.945763 -0.254773 -0.051217   
8099      0  1240247344131493888  0.034284 -0.945763 -0.254773 -0.051217   
8100      0  1271261017918472192  0.818228  0.185544  0.084825 -0.824511   
8101      0  1269957159724265472  0.818228  0.185544  0.084825 -0.824511   
8102      0  1269957159724265472  0.818228  0.185543  0.084825 -0.824511   

             4         5         6         7  ...  medicalnewstoday.com  \
0    -0.802968  0.731735  0.231155  0.420902  ...                     0   
1    -0.802968  0.731735  0.231155  0.420902  ...                     0   
2    -0.802968  0.731735  0.231155  0.420902  ...                     0   
3     0.145228  1.374485  0.138910 -0.195475  ...                     0   
4     0.145228  1.374485  0.138910 -0.195475  ...                     0   
...        ...       ...       ...       ...  ...                   ...   
8098  0.069432  0.803254 -0.061863  0.278749  ...                     1   
8099  0.069432  0.803254 -0.061863  0.278749  ...                     1   
8100 -0.313027  0.762159 -0.338960  0.064914  ...                     1   
8101 -0.313027  0.762159 -0.338960  0.064914  ...                     1   
8102 -0.313027  0.762159 -0.338960  0.064914  ...                     1   

      nih.gov  politifact.com  sciencedaily.com  twitter.com  webmd.com  \
0           0               0                 0            0          0   
1           0               0                 0            0          0   
2           0               0                 0            0          0   
3           0               0                 0            0          0   
4           0               0                 0            0          0   
...       ...             ...               ...          ...        ...   
8098        0               0                 0            0          0   
8099        0               0                 0            0          0   
8100        0               0                 0            0          0   
8101        0               0                 0            0          0   
8102        0               0                 0            0          0   

      who.int  www.factcheck.org  www.politifact.com  www.washingtonpost.com  
0           0                  0                   0                       0  
1           0                  0                   0                       0  
2           0                  0                   0                       0  
3           0                  0                   0                       0  
4           0                  0                   0                       0  
...       ...                ...                 ...                     ...  
8098        0                  0                   0                       0  
8099        0                  0                   0                       0  
8100        0                  0                   0                       0  
8101        0                  0                   0                       0  
8102        0                  0                   0                       0  

[8103 rows x 788 columns]

In [687]:
RealNodeFeature.to_csv('RealNodeFeature.csv')
FakeNodeFeature.to_csv('FakeNodeFeature.csv')
RealEdge.to_csv('RealEdge.csv')
FakeEdge.to_csv('FakeEdge.csv')